docs/modules/ROOT/pages/explanations/decisions/deletion-protection.adoc

```diff
@@ -0,0 +1,129 @@
+
+== Backend Proposal 2: Add additional logic to composition functions to ensure important objects aren't removed
+Have a composition function which checks at the end, some specified objects from the observed state are still present in the desired state.
+Either the function could simply add the observed object back, or simply abort the composition function with a fatal return.
+
+Advantages::
+* No actual deletion happens
+* No messing around with finalizers
+* No undelete process necessary, no deletion happened to begin with
+
+Disatvantages::
+* Increases complexity of the composition functions
+* Every service needs specific settings
+* Doesn't protect against deletion from outside AppCat
+* It's within composition functions itself
+
+== Backend Proposal 3: Deletion protection by custom validation webhook
+A validation webhook that will check on namespaces and pvcs.
+If a namespace/pvc belongs to an active AppCat instance, it will block the deletion of the object.
+An active AppCat instance is determined by the existence of the composite.
+So as long as the composite is not deleted, then the deletion of the namespace or pvc belonging to that composite will be blocked by the validation webhook.
+
+Advantages::
+* Can be implemented very generic, only core objects and metadata of composites are needed
+* Not part of the AppCat composition functions
+* No actual deletion happens
+* No messing around with finalizers
+* No undelete process necessary, no deletion happened to begin with
+
+Disatvantages::
+* No grace period time
+
+== Complementary Proposal: Use canary instances on the lab
+One instance of each service runs on the lab.
+Each new release will be deployed on the lab and tested extensively.
+After all tests succeed and no issues on the existing instances is observed, the release continues
+
+Advantages::
+* Some of the issues can be detected during testing on the lab
+* Rather easy to implement
+
+Disadvantages::
+* Would not have protected from the previous incident
+* More expensive, as lab resources would be used permanently
+
+== Decision
+
+**PROPOSED DECISION, NOT YET FINAL, NOT YET DISCUSSED WITH THE TEAM**
+
+* End-user Proposal 1: Protect claim deletion via webhooks
+* Backend Proposal 3: Deletion protection by custom validation webhook
+
+== Rationale
+While we loose the ability to delay the deletion with Webhooks, they are less complex to implement.
+Patching and managing finalizers on objects that are already managed by another controller leads to race conditions and bugs, as the current implementation shows.
+Webhooks also provide a much nicer user experience all around, as the user will get direct feedback from the system, if an invalid deletion command is issued.
+As the deletion will not hit the actual Kubernetes API, the `deletionTimestamp` is not set on the object, which effectively results in a no-op.
```

Comment:
Additional benefit:
GitOps issue - It's possible to create claim with `deletionProtection: false`, so for short living databases used in some kind of pipelines it should be really easy. For bigger ones which used to serve as a production instances, such protection is:
1) advantage - protects really important databases 
2) immediately visible - failed pipeline, terraform, kustomize or whatever else wanted to delete such an instance
3) easy to disable, by adding one additional step to pipeline if customer is double sure
win-win for me